src/module-elasticsuite-catalog/Model/ResourceModel/Eav/Indexer/Fulltext/Datasource/AbstractAttributeData.php

```diff
@@ -112,28 +112,43 @@ public function getAttributesRawData($storeId, array $entityIds, $tableName, arr
         // The legacy entity_id field.
         $entityIdField = $this->getEntityMetaData($this->getEntityTypeId())->getIdentifierField();
 
+        $joinDefaultValuesCondition = [
+            "entity.$linkField = t_default.$linkField",
+            $this->connection->quoteInto('t_default.attribute_id IN (?)', $attributeIds),
+            't_default.store_id= ?',
+        ];
+        $joinDefaultValuesCondition = $this->connection->quoteInto(
+            implode(' AND ', $joinDefaultValuesCondition),
+            \Magento\Store\Model\Store::DEFAULT_STORE_ID
+        );
+
         $joinStoreValuesConditionClauses = [
-            "t_default.$linkField = t_store.$linkField",
-            't_default.attribute_id = t_store.attribute_id',
+            "entity.$linkField = t_store.$linkField",
+            $this->connection->quoteInto('t_store.attribute_id IN (?)', $attributeIds),
             't_store.store_id= ?',
         ];
-
         $joinStoreValuesCondition = $this->connection->quoteInto(
             implode(' AND ', $joinStoreValuesConditionClauses),
             $storeId
         );
 
         $select->from(['entity' => $this->getEntityMetaData($this->getEntityTypeId())->getEntityTable()], [$entityIdField])
-            ->joinInner(
+            ->joinLeft(
                 ['t_default' => $tableName],
-                new \Zend_Db_Expr("entity.{$linkField} = t_default.{$linkField}"),
-                ['attribute_id']
+                $joinDefaultValuesCondition,
+                []
+            )
+            ->joinLeft(
+                ['t_store' => $tableName],
+                $joinStoreValuesCondition,
+                []
             )
-            ->joinLeft(['t_store' => $tableName], $joinStoreValuesCondition, [])
-            ->where('t_default.store_id=?', 0)
-            ->where('t_default.attribute_id IN (?)', $attributeIds)
             ->where("entity.{$entityIdField} IN (?)", $entityIds)
-            ->columns(['value' => new \Zend_Db_Expr('COALESCE(t_store.value, t_default.value)')]);
+            ->having('value IS NOT NULL')
```

Comment:
What is the purpose of that HAVING clause, since we are using COALESCE ?
Which "value" does it represent here ? the default scope one or the store scope one ? or both ?

Reply:
OK, it's for there is no record for the attribute_id/entity_id in the table. It prevents an index notice for categories on Luma.
But I'll will probably change the condition for 'attribute_id' instead of value : null for 'value' is usually ignored by `\Smile\ElasticsuiteCatalog\Helper\AbstractAttribute::prepareIndexValue`.